In [287]:
import pandas as pd

In [288]:
# Checking the version of Pandas
pd.__version__

'3.0.1'

In [289]:
# importing the csv diabetes xpt dataset
diabetes = pd.read_sas("BMX_J.xpt")
diabetes

,SEQN,BMDSTATS,BMXWT,BMIWT,BMXRECUM,BMIRECUM,BMXHEAD,BMIHEAD,BMXHT,BMIHT,...,BMXLEG,BMILEG,BMXARML,BMIARML,BMXARMC,BMIARMC,BMXWAIST,BMIWAIST,BMXHIP,BMIHIP
0,93703.0,1.0,13.7,3.0,89.6,NaN,NaN,NaN,88.6,NaN,...,NaN,NaN,18.0,NaN,16.2,NaN,48.2,NaN,NaN,NaN
1,93704.0,1.0,13.9,NaN,95.0,NaN,NaN,NaN,94.2,NaN,...,NaN,NaN,18.6,NaN,15.2,NaN,50.0,NaN,NaN,NaN
2,93705.0,1.0,79.5,NaN,NaN,NaN,NaN,NaN,158.3,NaN,...,37.0,NaN,36.0,NaN,32.0,NaN,101.8,NaN,110.0,NaN
3,93706.0,1.0,66.3,NaN,NaN,NaN,NaN,NaN,175.7,NaN,...,46.6,NaN,38.8,NaN,27.0,NaN,79.3,NaN,94.4,NaN
4,93707.0,1.0,45.4,NaN,NaN,NaN,NaN,NaN,158.4,NaN,...,38.1,NaN,33.8,NaN,21.5,NaN,64.1,NaN,83.0,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
8699,102952.0,1.0,49.0,NaN,NaN,NaN,NaN,NaN,156.5,NaN,...,34.4,NaN,32.6,NaN,25.1,NaN,82.2,NaN,87.3,NaN
8700,102953.0,1.0,97.4,NaN,NaN,NaN,NaN,NaN,164.9,NaN,...,38.2,NaN,36.6,NaN,40.6,NaN,114.8,NaN,112.8,NaN
8701,102954.0,1.0,69.1,NaN,NaN,NaN,NaN,NaN,162.6,NaN,...,39.2,NaN,35.2,NaN,26.8,NaN,86.4,NaN,102.7,NaN
8702,102955.0,1.0,111.9,NaN,NaN,NaN,NaN,NaN,156.6,NaN,...,39.2,NaN,35.0,NaN,44.5,NaN,113.5,NaN,128.3,NaN


In [290]:
diabetes.isnull().sum()

SEQN           0
BMDSTATS       0
BMXWT        124
BMIWT       8288
BMXRECUM    7810
BMIRECUM    8680
BMXHEAD     8510
BMIHEAD     8704
BMXHT        688
BMIHT       8605
BMXBMI       699
BMXLEG      2001
BMILEG      8370
BMXARML      527
BMIARML     8357
BMXARMC      531
BMIARMC     8354
BMXWAIST    1103
BMIWAIST    8267
BMXHIP      2665
BMIHIP      8434
dtype: int64

In [291]:
diabetes.info()

<class 'pandas.DataFrame'>
RangeIndex: 8704 entries, 0 to 8703
Data columns (total 21 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   SEQN      8704 non-null   float64
 1   BMDSTATS  8704 non-null   float64
 2   BMXWT     8580 non-null   float64
 3   BMIWT     416 non-null    float64
 4   BMXRECUM  894 non-null    float64
 5   BMIRECUM  24 non-null     float64
 6   BMXHEAD   194 non-null    float64
 7   BMIHEAD   0 non-null      float64
 8   BMXHT     8016 non-null   float64
 9   BMIHT     99 non-null     float64
 10  BMXBMI    8005 non-null   float64
 11  BMXLEG    6703 non-null   float64
 12  BMILEG    334 non-null    float64
 13  BMXARML   8177 non-null   float64
 14  BMIARML   347 non-null    float64
 15  BMXARMC   8173 non-null   float64
 16  BMIARMC   350 non-null    float64
 17  BMXWAIST  7601 non-null   float64
 18  BMIWAIST  437 non-null    float64
 19  BMXHIP    6039 non-null   float64
 20  BMIHIP    270 non-null    float64
dtypes:

In [292]:
diabetes_1 = pd.read_sas("DIQ_J.xpt")
diabetes_1

,SEQN,DIQ010,DID040,DIQ160,DIQ170,DIQ172,DIQ175A,DIQ175B,DIQ175C,DIQ175D,...,DIQ300D,DID310S,DID310D,DID320,DID330,DID341,DID350,DIQ350U,DIQ360,DIQ080
0,93703.0,2.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,93704.0,2.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,93705.0,2.0,NaN,2.0,2.0,2.0,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,93706.0,2.0,NaN,2.0,2.0,2.0,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,93707.0,2.0,NaN,2.0,2.0,2.0,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
8892,102952.0,3.0,NaN,NaN,2.0,2.0,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
8893,102953.0,2.0,NaN,1.0,2.0,1.0,NaN,11.0,NaN,13.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
8894,102954.0,2.0,NaN,2.0,2.0,2.0,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
8895,102955.0,2.0,NaN,2.0,2.0,1.0,NaN,11.0,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [293]:
diabetes_1.info()

<class 'pandas.DataFrame'>
RangeIndex: 8897 entries, 0 to 8896
Data columns (total 54 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   SEQN     8897 non-null   float64
 1   DIQ010   8897 non-null   float64
 2   DID040   892 non-null    float64
 3   DIQ160   5694 non-null   float64
 4   DIQ170   5876 non-null   float64
 5   DIQ172   5876 non-null   float64
 6   DIQ175A  1145 non-null   float64
 7   DIQ175B  657 non-null    float64
 8   DIQ175C  217 non-null    float64
 9   DIQ175D  488 non-null    float64
 10  DIQ175E  177 non-null    float64
 11  DIQ175F  38 non-null     float64
 12  DIQ175G  318 non-null    float64
 13  DIQ175H  291 non-null    float64
 14  DIQ175I  76 non-null     float64
 15  DIQ175J  163 non-null    float64
 16  DIQ175K  29 non-null     float64
 17  DIQ175L  71 non-null     float64
 18  DIQ175M  155 non-null    float64
 19  DIQ175N  113 non-null    float64
 20  DIQ175O  167 non-null    float64
 21  DIQ175P  143 non-null    

In [294]:
diabetes_1.isnull().sum()

SEQN          0
DIQ010        0
DID040     8005
DIQ160     3203
DIQ170     3021
DIQ172     3021
DIQ175A    7752
DIQ175B    8240
DIQ175C    8680
DIQ175D    8409
DIQ175E    8720
DIQ175F    8859
DIQ175G    8579
DIQ175H    8606
DIQ175I    8821
DIQ175J    8734
DIQ175K    8868
DIQ175L    8826
DIQ175M    8742
DIQ175N    8784
DIQ175O    8730
DIQ175P    8754
DIQ175Q    8725
DIQ175R    8878
DIQ175S    8860
DIQ175T    8781
DIQ175U    8784
DIQ175V    8883
DIQ175W    8892
DIQ175X    8895
DIQ180     3021
DIQ050        0
DID060     8643
DIQ060U    8652
DIQ070     7223
DIQ230     8005
DIQ240     8005
DID250     8203
DID260     8004
DIQ260U    8184
DIQ275     8005
DIQ280     8150
DIQ291     8148
DIQ300S    8013
DIQ300D    8013
DID310S    8011
DID310D    8011
DID320     8012
DID330     8066
DID341     8015
DID350     8015
DIQ350U    8185
DIQ360     8012
DIQ080     8012
dtype: int64

The high volume of missing (NaN) values across most columns would have significantly reduced the quality and reliability of the data, making them unsuitable for building an accurate machine learning model.

In [295]:
# Loading maternity health dataset
maternity = pd.read_csv("Maternal Health Risk Data Set.csv")
maternity

,Age,SystolicBP,DiastolicBP,BS,BodyTemp,HeartRate,RiskLevel
0,25,130,80,15.0,98.0,86,high risk
1,35,140,90,13.0,98.0,70,high risk
2,29,90,70,8.0,100.0,80,high risk
3,30,140,85,7.0,98.0,70,high risk
4,35,120,60,6.1,98.0,76,low risk
...,...,...,...,...,...,...,...
1009,22,120,60,15.0,98.0,80,high risk
1010,55,120,90,18.0,98.0,60,high risk
1011,35,85,60,19.0,98.0,86,high risk
1012,43,120,90,18.0,98.0,70,high risk


In [296]:
# Checking all rows in maternity health dataset
print(maternity.to_string())

      Age  SystolicBP  DiastolicBP     BS  BodyTemp  HeartRate  RiskLevel
0      25         130           80  15.00      98.0         86  high risk
1      35         140           90  13.00      98.0         70  high risk
2      29          90           70   8.00     100.0         80  high risk
3      30         140           85   7.00      98.0         70  high risk
4      35         120           60   6.10      98.0         76   low risk
5      23         140           80   7.01      98.0         70  high risk
6      23         130           70   7.01      98.0         78   mid risk
7      35          85           60  11.00     102.0         86  high risk
8      32         120           90   6.90      98.0         70   mid risk
9      42         130           80  18.00      98.0         70  high risk
10     23          90           60   7.01      98.0         76   low risk
11     19         120           80   7.00      98.0         70   mid risk
12     25         110           89   7

In [297]:
# Checking maternity health data information
maternity.info()

<class 'pandas.DataFrame'>
RangeIndex: 1014 entries, 0 to 1013
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Age          1014 non-null   int64  
 1   SystolicBP   1014 non-null   int64  
 2   DiastolicBP  1014 non-null   int64  
 3   BS           1014 non-null   float64
 4   BodyTemp     1014 non-null   float64
 5   HeartRate    1014 non-null   int64  
 6   RiskLevel    1014 non-null   str    
dtypes: float64(2), int64(4), str(1)
memory usage: 55.6 KB


In [298]:
# Checking for null values
maternity.isnull().sum()

Age            0
SystolicBP     0
DiastolicBP    0
BS             0
BodyTemp       0
HeartRate      0
RiskLevel      0
dtype: int64

In [299]:
# Checking the dimensions of the dataset 
maternity.shape

(1014, 7)

In [300]:
# Checking for duplicate values
print(maternity.duplicated().sum())

562


In [301]:
# Making a copy of the original data
maternity_risk = maternity.copy()
maternity_risk

,Age,SystolicBP,DiastolicBP,BS,BodyTemp,HeartRate,RiskLevel
0,25,130,80,15.0,98.0,86,high risk
1,35,140,90,13.0,98.0,70,high risk
2,29,90,70,8.0,100.0,80,high risk
3,30,140,85,7.0,98.0,70,high risk
4,35,120,60,6.1,98.0,76,low risk
...,...,...,...,...,...,...,...
1009,22,120,60,15.0,98.0,80,high risk
1010,55,120,90,18.0,98.0,60,high risk
1011,35,85,60,19.0,98.0,86,high risk
1012,43,120,90,18.0,98.0,70,high risk


In [302]:
# Confirming duplicate values
print(maternity_risk.duplicated().sum())

562


In [303]:
# # Dropping duplicate values
maternity_risk.drop_duplicates(inplace = True)
maternity_risk

,Age,SystolicBP,DiastolicBP,BS,BodyTemp,HeartRate,RiskLevel
0,25,130,80,15.0,98.0,86,high risk
1,35,140,90,13.0,98.0,70,high risk
2,29,90,70,8.0,100.0,80,high risk
3,30,140,85,7.0,98.0,70,high risk
4,35,120,60,6.1,98.0,76,low risk
...,...,...,...,...,...,...,...
673,12,100,50,6.4,98.0,70,mid risk
674,15,100,60,6.0,98.0,80,low risk
703,15,100,49,7.6,98.0,77,low risk
704,12,100,50,6.0,98.0,70,mid risk


In [304]:
print(maternity_risk.duplicated().sum())

0


In [305]:
# Confirming the dimension of the new data
maternity_risk.shape

(452, 7)

In [306]:
# Checking the number of columns
print(maternity_risk.columns.to_list())

['Age', 'SystolicBP', 'DiastolicBP', 'BS', 'BodyTemp', 'HeartRate', 'RiskLevel']


In [307]:
# Checking the number of unique values
maternity_risk.nunique()

Age            50
SystolicBP     19
DiastolicBP    16
BS             29
BodyTemp        8
HeartRate      16
RiskLevel       3
dtype: int64

In [308]:
# using expression aproach to check categorical data features
features = [col for col in maternity_risk.columns if maternity_risk[col].nunique() < 10 and maternity_risk[col].nunique() > 1]
print(features)

['BodyTemp', 'RiskLevel']


In [309]:
# Checking unique values in SystolicBP, DiastolicBP, BodyTemp, HeartRate and RiskLevel
print(maternity_risk["BodyTemp"].unique())
print(maternity_risk["RiskLevel"].unique())

[ 98.  100.  102.  101.  103.   98.4  99.   98.6]
<StringArray>
['high risk', 'low risk', 'mid risk']
Length: 3, dtype: str


In [310]:
# Changing the data type of features to category
maternity_risk[features] = maternity_risk[features].astype("category")
maternity_risk.info()

<class 'pandas.DataFrame'>
Index: 452 entries, 0 to 705
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   Age          452 non-null    int64   
 1   SystolicBP   452 non-null    int64   
 2   DiastolicBP  452 non-null    int64   
 3   BS           452 non-null    float64 
 4   BodyTemp     452 non-null    category
 5   HeartRate    452 non-null    int64   
 6   RiskLevel    452 non-null    category
dtypes: category(2), float64(1), int64(4)
memory usage: 22.2 KB


In [311]:
maternity_risk.shape

(452, 7)

The dataset originally contained 1,014 rows and 7 columns. A new copy of the dataset was created, after removing 562 duplicate rows in the newly ceated copy, the dataset was reduced to 452 rows. No missing values were found. Two columns(BodyTemp and RiskLevel) were converted to category type. 

In [312]:
# Checking the fetal health datatset
fetal = pd.read_csv("fetal_health.csv")
fetal

,baseline value,accelerations,fetal_movement,uterine_contractions,light_decelerations,severe_decelerations,prolongued_decelerations,abnormal_short_term_variability,mean_value_of_short_term_variability,percentage_of_time_with_abnormal_long_term_variability,...,histogram_min,histogram_max,histogram_number_of_peaks,histogram_number_of_zeroes,histogram_mode,histogram_mean,histogram_median,histogram_variance,histogram_tendency,fetal_health
0,120.0,0.000,0.000,0.000,0.000,0.0,0.0,73.0,0.5,43.0,...,62.0,126.0,2.0,0.0,120.0,137.0,121.0,73.0,1.0,2.0
1,132.0,0.006,0.000,0.006,0.003,0.0,0.0,17.0,2.1,0.0,...,68.0,198.0,6.0,1.0,141.0,136.0,140.0,12.0,0.0,1.0
2,133.0,0.003,0.000,0.008,0.003,0.0,0.0,16.0,2.1,0.0,...,68.0,198.0,5.0,1.0,141.0,135.0,138.0,13.0,0.0,1.0
3,134.0,0.003,0.000,0.008,0.003,0.0,0.0,16.0,2.4,0.0,...,53.0,170.0,11.0,0.0,137.0,134.0,137.0,13.0,1.0,1.0
4,132.0,0.007,0.000,0.008,0.000,0.0,0.0,16.0,2.4,0.0,...,53.0,170.0,9.0,0.0,137.0,136.0,138.0,11.0,1.0,1.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2121,140.0,0.000,0.000,0.007,0.000,0.0,0.0,79.0,0.2,25.0,...,137.0,177.0,4.0,0.0,153.0,150.0,152.0,2.0,0.0,2.0
2122,140.0,0.001,0.000,0.007,0.000,0.0,0.0,78.0,0.4,22.0,...,103.0,169.0,6.0,0.0,152.0,148.0,151.0,3.0,1.0,2.0
2123,140.0,0.001,0.000,0.007,0.000,0.0,0.0,79.0,0.4,20.0,...,103.0,170.0,5.0,0.0,153.0,148.0,152.0,4.0,1.0,2.0
2124,140.0,0.001,0.000,0.006,0.000,0.0,0.0,78.0,0.4,27.0,...,103.0,169.0,6.0,0.0,152.0,147.0,151.0,4.0,1.0,2.0


In [313]:
# Checking 
fetal.info()

<class 'pandas.DataFrame'>
RangeIndex: 2126 entries, 0 to 2125
Data columns (total 22 columns):
 #   Column                                                  Non-Null Count  Dtype  
---  ------                                                  --------------  -----  
 0   baseline value                                          2126 non-null   float64
 1   accelerations                                           2126 non-null   float64
 2   fetal_movement                                          2126 non-null   float64
 3   uterine_contractions                                    2126 non-null   float64
 4   light_decelerations                                     2126 non-null   float64
 5   severe_decelerations                                    2126 non-null   float64
 6   prolongued_decelerations                                2126 non-null   float64
 7   abnormal_short_term_variability                         2126 non-null   float64
 8   mean_value_of_short_term_variability             

In [314]:
# Checking the dimension 
fetal.shape

(2126, 22)

In [315]:
# Checking the for null values in the dataset
fetal.isnull().sum()

baseline value                                            0
accelerations                                             0
fetal_movement                                            0
uterine_contractions                                      0
light_decelerations                                       0
severe_decelerations                                      0
prolongued_decelerations                                  0
abnormal_short_term_variability                           0
mean_value_of_short_term_variability                      0
percentage_of_time_with_abnormal_long_term_variability    0
mean_value_of_long_term_variability                       0
histogram_width                                           0
histogram_min                                             0
histogram_max                                             0
histogram_number_of_peaks                                 0
histogram_number_of_zeroes                                0
histogram_mode                          

In [316]:
# Checking for duplicate values
print(fetal.duplicated().sum())

13


In [317]:
# Copying the original dataset
fetal_risk = fetal.copy()
fetal_risk

,baseline value,accelerations,fetal_movement,uterine_contractions,light_decelerations,severe_decelerations,prolongued_decelerations,abnormal_short_term_variability,mean_value_of_short_term_variability,percentage_of_time_with_abnormal_long_term_variability,...,histogram_min,histogram_max,histogram_number_of_peaks,histogram_number_of_zeroes,histogram_mode,histogram_mean,histogram_median,histogram_variance,histogram_tendency,fetal_health
0,120.0,0.000,0.000,0.000,0.000,0.0,0.0,73.0,0.5,43.0,...,62.0,126.0,2.0,0.0,120.0,137.0,121.0,73.0,1.0,2.0
1,132.0,0.006,0.000,0.006,0.003,0.0,0.0,17.0,2.1,0.0,...,68.0,198.0,6.0,1.0,141.0,136.0,140.0,12.0,0.0,1.0
2,133.0,0.003,0.000,0.008,0.003,0.0,0.0,16.0,2.1,0.0,...,68.0,198.0,5.0,1.0,141.0,135.0,138.0,13.0,0.0,1.0
3,134.0,0.003,0.000,0.008,0.003,0.0,0.0,16.0,2.4,0.0,...,53.0,170.0,11.0,0.0,137.0,134.0,137.0,13.0,1.0,1.0
4,132.0,0.007,0.000,0.008,0.000,0.0,0.0,16.0,2.4,0.0,...,53.0,170.0,9.0,0.0,137.0,136.0,138.0,11.0,1.0,1.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2121,140.0,0.000,0.000,0.007,0.000,0.0,0.0,79.0,0.2,25.0,...,137.0,177.0,4.0,0.0,153.0,150.0,152.0,2.0,0.0,2.0
2122,140.0,0.001,0.000,0.007,0.000,0.0,0.0,78.0,0.4,22.0,...,103.0,169.0,6.0,0.0,152.0,148.0,151.0,3.0,1.0,2.0
2123,140.0,0.001,0.000,0.007,0.000,0.0,0.0,79.0,0.4,20.0,...,103.0,170.0,5.0,0.0,153.0,148.0,152.0,4.0,1.0,2.0
2124,140.0,0.001,0.000,0.006,0.000,0.0,0.0,78.0,0.4,27.0,...,103.0,169.0,6.0,0.0,152.0,147.0,151.0,4.0,1.0,2.0


In [318]:
# Dropping duplicate values
fetal_risk.drop_duplicates(inplace = True)
fetal_risk

,baseline value,accelerations,fetal_movement,uterine_contractions,light_decelerations,severe_decelerations,prolongued_decelerations,abnormal_short_term_variability,mean_value_of_short_term_variability,percentage_of_time_with_abnormal_long_term_variability,...,histogram_min,histogram_max,histogram_number_of_peaks,histogram_number_of_zeroes,histogram_mode,histogram_mean,histogram_median,histogram_variance,histogram_tendency,fetal_health
0,120.0,0.000,0.000,0.000,0.000,0.0,0.0,73.0,0.5,43.0,...,62.0,126.0,2.0,0.0,120.0,137.0,121.0,73.0,1.0,2.0
1,132.0,0.006,0.000,0.006,0.003,0.0,0.0,17.0,2.1,0.0,...,68.0,198.0,6.0,1.0,141.0,136.0,140.0,12.0,0.0,1.0
2,133.0,0.003,0.000,0.008,0.003,0.0,0.0,16.0,2.1,0.0,...,68.0,198.0,5.0,1.0,141.0,135.0,138.0,13.0,0.0,1.0
3,134.0,0.003,0.000,0.008,0.003,0.0,0.0,16.0,2.4,0.0,...,53.0,170.0,11.0,0.0,137.0,134.0,137.0,13.0,1.0,1.0
4,132.0,0.007,0.000,0.008,0.000,0.0,0.0,16.0,2.4,0.0,...,53.0,170.0,9.0,0.0,137.0,136.0,138.0,11.0,1.0,1.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2121,140.0,0.000,0.000,0.007,0.000,0.0,0.0,79.0,0.2,25.0,...,137.0,177.0,4.0,0.0,153.0,150.0,152.0,2.0,0.0,2.0
2122,140.0,0.001,0.000,0.007,0.000,0.0,0.0,78.0,0.4,22.0,...,103.0,169.0,6.0,0.0,152.0,148.0,151.0,3.0,1.0,2.0
2123,140.0,0.001,0.000,0.007,0.000,0.0,0.0,79.0,0.4,20.0,...,103.0,170.0,5.0,0.0,153.0,148.0,152.0,4.0,1.0,2.0
2124,140.0,0.001,0.000,0.006,0.000,0.0,0.0,78.0,0.4,27.0,...,103.0,169.0,6.0,0.0,152.0,147.0,151.0,4.0,1.0,2.0


In [319]:
# Checking the dimension 
fetal_risk.shape

(2113, 22)

In [320]:
# Checking the number of column
print(fetal_risk.columns)

Index(['baseline value', 'accelerations', 'fetal_movement',
       'uterine_contractions', 'light_decelerations', 'severe_decelerations',
       'prolongued_decelerations', 'abnormal_short_term_variability',
       'mean_value_of_short_term_variability',
       'percentage_of_time_with_abnormal_long_term_variability',
       'mean_value_of_long_term_variability', 'histogram_width',
       'histogram_min', 'histogram_max', 'histogram_number_of_peaks',
       'histogram_number_of_zeroes', 'histogram_mode', 'histogram_mean',
       'histogram_median', 'histogram_variance', 'histogram_tendency',
       'fetal_health'],
      dtype='str')


In [321]:
# Checking the number of unique features
fetal_risk.nunique()

baseline value                                             48
accelerations                                              20
fetal_movement                                            102
uterine_contractions                                       16
light_decelerations                                        16
severe_decelerations                                        2
prolongued_decelerations                                    6
abnormal_short_term_variability                            75
mean_value_of_short_term_variability                       57
percentage_of_time_with_abnormal_long_term_variability     87
mean_value_of_long_term_variability                       249
histogram_width                                           154
histogram_min                                             109
histogram_max                                              86
histogram_number_of_peaks                                  18
histogram_number_of_zeroes                                  9
histogra

In [322]:
# Expression approach for checking categorical features
features = [col for col in fetal_risk.columns if fetal_risk[col].nunique() <  10 and fetal_risk[col].nunique() > 1]
print(features)

['severe_decelerations', 'prolongued_decelerations', 'histogram_number_of_zeroes', 'histogram_tendency', 'fetal_health']


In [323]:
print(fetal_risk["severe_decelerations"].unique())
print(fetal_risk["prolongued_decelerations"].unique())
print(fetal_risk["histogram_number_of_zeroes"].unique())
print(fetal_risk["histogram_tendency"].unique())
print(fetal_risk["fetal_health"].unique())
# print(fetal_risk["alcohol_consumption_per_week"].unique())
# print(fetal_risk["smoking_status"].unique())
# print(diabetes_dataset_1["income_level"].unique())
# print(diabetes_dataset_1["ethnicity"].unique())

[0.    0.001]
[0.    0.002 0.003 0.001 0.004 0.005]
[ 0.  1.  3.  2.  4. 10.  5.  8.  7.]
[ 1.  0. -1.]
[2. 1. 3.]


In [324]:
# Changing features to category
fetal_risk[features] = fetal_risk[features].astype("category")
fetal_risk.info()

<class 'pandas.DataFrame'>
Index: 2113 entries, 0 to 2125
Data columns (total 22 columns):
 #   Column                                                  Non-Null Count  Dtype   
---  ------                                                  --------------  -----   
 0   baseline value                                          2113 non-null   float64 
 1   accelerations                                           2113 non-null   float64 
 2   fetal_movement                                          2113 non-null   float64 
 3   uterine_contractions                                    2113 non-null   float64 
 4   light_decelerations                                     2113 non-null   float64 
 5   severe_decelerations                                    2113 non-null   category
 6   prolongued_decelerations                                2113 non-null   category
 7   abnormal_short_term_variability                         2113 non-null   float64 
 8   mean_value_of_short_term_variability        

The dataset originally contained 2,126 rows and 22 columns. A new copy of the dataeset was created, after removing 13 duplicate rows in the newly created copy, the dataset was reduced to 2,113 rows. No missing values were found. Five columns(severe_decelerations, prolongued_decelerations, histogram_number_of_zeroes, histogram_tendency, and fetal_health) were converted to category type.

In [325]:
# importing the csv dataset
diabetes_prediction = pd.read_csv("diabetes_012_health_indicators_BRFSS2015.csv")
diabetes_prediction

,Diabetes_012,HighBP,HighChol,CholCheck,BMI,Smoker,Stroke,HeartDiseaseorAttack,PhysActivity,Fruits,...,AnyHealthcare,NoDocbcCost,GenHlth,MentHlth,PhysHlth,DiffWalk,Sex,Age,Education,Income
0,0.0,1.0,1.0,1.0,40.0,1.0,0.0,0.0,0.0,0.0,...,1.0,0.0,5.0,18.0,15.0,1.0,0.0,9.0,4.0,3.0
1,0.0,0.0,0.0,0.0,25.0,1.0,0.0,0.0,1.0,0.0,...,0.0,1.0,3.0,0.0,0.0,0.0,0.0,7.0,6.0,1.0
2,0.0,1.0,1.0,1.0,28.0,0.0,0.0,0.0,0.0,1.0,...,1.0,1.0,5.0,30.0,30.0,1.0,0.0,9.0,4.0,8.0
3,0.0,1.0,0.0,1.0,27.0,0.0,0.0,0.0,1.0,1.0,...,1.0,0.0,2.0,0.0,0.0,0.0,0.0,11.0,3.0,6.0
4,0.0,1.0,1.0,1.0,24.0,0.0,0.0,0.0,1.0,1.0,...,1.0,0.0,2.0,3.0,0.0,0.0,0.0,11.0,5.0,4.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
253675,0.0,1.0,1.0,1.0,45.0,0.0,0.0,0.0,0.0,1.0,...,1.0,0.0,3.0,0.0,5.0,0.0,1.0,5.0,6.0,7.0
253676,2.0,1.0,1.0,1.0,18.0,0.0,0.0,0.0,0.0,0.0,...,1.0,0.0,4.0,0.0,0.0,1.0,0.0,11.0,2.0,4.0
253677,0.0,0.0,0.0,1.0,28.0,0.0,0.0,0.0,1.0,1.0,...,1.0,0.0,1.0,0.0,0.0,0.0,0.0,2.0,5.0,2.0
253678,0.0,1.0,0.0,1.0,23.0,0.0,0.0,0.0,0.0,1.0,...,1.0,0.0,3.0,0.0,0.0,0.0,1.0,7.0,5.0,1.0


In [326]:
# Checking the dataset information
diabetes_prediction.info()

<class 'pandas.DataFrame'>
RangeIndex: 253680 entries, 0 to 253679
Data columns (total 22 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   Diabetes_012          253680 non-null  float64
 1   HighBP                253680 non-null  float64
 2   HighChol              253680 non-null  float64
 3   CholCheck             253680 non-null  float64
 4   BMI                   253680 non-null  float64
 5   Smoker                253680 non-null  float64
 6   Stroke                253680 non-null  float64
 7   HeartDiseaseorAttack  253680 non-null  float64
 8   PhysActivity          253680 non-null  float64
 9   Fruits                253680 non-null  float64
 10  Veggies               253680 non-null  float64
 11  HvyAlcoholConsump     253680 non-null  float64
 12  AnyHealthcare         253680 non-null  float64
 13  NoDocbcCost           253680 non-null  float64
 14  GenHlth               253680 non-null  float64
 15  MentHlth   

In [327]:
# Checking the dimension
diabetes_prediction.shape

(253680, 22)

In [328]:
# Checking for null values
diabetes_prediction.isnull().sum()

Diabetes_012            0
HighBP                  0
HighChol                0
CholCheck               0
BMI                     0
Smoker                  0
Stroke                  0
HeartDiseaseorAttack    0
PhysActivity            0
Fruits                  0
Veggies                 0
HvyAlcoholConsump       0
AnyHealthcare           0
NoDocbcCost             0
GenHlth                 0
MentHlth                0
PhysHlth                0
DiffWalk                0
Sex                     0
Age                     0
Education               0
Income                  0
dtype: int64

In [329]:
# Checking for duplicate values
print(diabetes_prediction.duplicated().sum())

23899


In [330]:
# Copying the original dataset
diabetes_prediction_2 = diabetes_prediction.copy()
diabetes_prediction_2

,Diabetes_012,HighBP,HighChol,CholCheck,BMI,Smoker,Stroke,HeartDiseaseorAttack,PhysActivity,Fruits,...,AnyHealthcare,NoDocbcCost,GenHlth,MentHlth,PhysHlth,DiffWalk,Sex,Age,Education,Income
0,0.0,1.0,1.0,1.0,40.0,1.0,0.0,0.0,0.0,0.0,...,1.0,0.0,5.0,18.0,15.0,1.0,0.0,9.0,4.0,3.0
1,0.0,0.0,0.0,0.0,25.0,1.0,0.0,0.0,1.0,0.0,...,0.0,1.0,3.0,0.0,0.0,0.0,0.0,7.0,6.0,1.0
2,0.0,1.0,1.0,1.0,28.0,0.0,0.0,0.0,0.0,1.0,...,1.0,1.0,5.0,30.0,30.0,1.0,0.0,9.0,4.0,8.0
3,0.0,1.0,0.0,1.0,27.0,0.0,0.0,0.0,1.0,1.0,...,1.0,0.0,2.0,0.0,0.0,0.0,0.0,11.0,3.0,6.0
4,0.0,1.0,1.0,1.0,24.0,0.0,0.0,0.0,1.0,1.0,...,1.0,0.0,2.0,3.0,0.0,0.0,0.0,11.0,5.0,4.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
253675,0.0,1.0,1.0,1.0,45.0,0.0,0.0,0.0,0.0,1.0,...,1.0,0.0,3.0,0.0,5.0,0.0,1.0,5.0,6.0,7.0
253676,2.0,1.0,1.0,1.0,18.0,0.0,0.0,0.0,0.0,0.0,...,1.0,0.0,4.0,0.0,0.0,1.0,0.0,11.0,2.0,4.0
253677,0.0,0.0,0.0,1.0,28.0,0.0,0.0,0.0,1.0,1.0,...,1.0,0.0,1.0,0.0,0.0,0.0,0.0,2.0,5.0,2.0
253678,0.0,1.0,0.0,1.0,23.0,0.0,0.0,0.0,0.0,1.0,...,1.0,0.0,3.0,0.0,0.0,0.0,1.0,7.0,5.0,1.0


In [331]:
# Dropping all duplicates
diabetes_prediction_2.drop_duplicates(inplace = True)
diabetes_prediction_2

,Diabetes_012,HighBP,HighChol,CholCheck,BMI,Smoker,Stroke,HeartDiseaseorAttack,PhysActivity,Fruits,...,AnyHealthcare,NoDocbcCost,GenHlth,MentHlth,PhysHlth,DiffWalk,Sex,Age,Education,Income
0,0.0,1.0,1.0,1.0,40.0,1.0,0.0,0.0,0.0,0.0,...,1.0,0.0,5.0,18.0,15.0,1.0,0.0,9.0,4.0,3.0
1,0.0,0.0,0.0,0.0,25.0,1.0,0.0,0.0,1.0,0.0,...,0.0,1.0,3.0,0.0,0.0,0.0,0.0,7.0,6.0,1.0
2,0.0,1.0,1.0,1.0,28.0,0.0,0.0,0.0,0.0,1.0,...,1.0,1.0,5.0,30.0,30.0,1.0,0.0,9.0,4.0,8.0
3,0.0,1.0,0.0,1.0,27.0,0.0,0.0,0.0,1.0,1.0,...,1.0,0.0,2.0,0.0,0.0,0.0,0.0,11.0,3.0,6.0
4,0.0,1.0,1.0,1.0,24.0,0.0,0.0,0.0,1.0,1.0,...,1.0,0.0,2.0,3.0,0.0,0.0,0.0,11.0,5.0,4.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
253675,0.0,1.0,1.0,1.0,45.0,0.0,0.0,0.0,0.0,1.0,...,1.0,0.0,3.0,0.0,5.0,0.0,1.0,5.0,6.0,7.0
253676,2.0,1.0,1.0,1.0,18.0,0.0,0.0,0.0,0.0,0.0,...,1.0,0.0,4.0,0.0,0.0,1.0,0.0,11.0,2.0,4.0
253677,0.0,0.0,0.0,1.0,28.0,0.0,0.0,0.0,1.0,1.0,...,1.0,0.0,1.0,0.0,0.0,0.0,0.0,2.0,5.0,2.0
253678,0.0,1.0,0.0,1.0,23.0,0.0,0.0,0.0,0.0,1.0,...,1.0,0.0,3.0,0.0,0.0,0.0,1.0,7.0,5.0,1.0


In [332]:
# Dropping columns
diabetes_prediction_2.drop(columns = ["Education", "Sex"], inplace=True)
diabetes_prediction_2

,Diabetes_012,HighBP,HighChol,CholCheck,BMI,Smoker,Stroke,HeartDiseaseorAttack,PhysActivity,Fruits,Veggies,HvyAlcoholConsump,AnyHealthcare,NoDocbcCost,GenHlth,MentHlth,PhysHlth,DiffWalk,Age,Income
0,0.0,1.0,1.0,1.0,40.0,1.0,0.0,0.0,0.0,0.0,1.0,0.0,1.0,0.0,5.0,18.0,15.0,1.0,9.0,3.0
1,0.0,0.0,0.0,0.0,25.0,1.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,1.0,3.0,0.0,0.0,0.0,7.0,1.0
2,0.0,1.0,1.0,1.0,28.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,1.0,1.0,5.0,30.0,30.0,1.0,9.0,8.0
3,0.0,1.0,0.0,1.0,27.0,0.0,0.0,0.0,1.0,1.0,1.0,0.0,1.0,0.0,2.0,0.0,0.0,0.0,11.0,6.0
4,0.0,1.0,1.0,1.0,24.0,0.0,0.0,0.0,1.0,1.0,1.0,0.0,1.0,0.0,2.0,3.0,0.0,0.0,11.0,4.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
253675,0.0,1.0,1.0,1.0,45.0,0.0,0.0,0.0,0.0,1.0,1.0,0.0,1.0,0.0,3.0,0.0,5.0,0.0,5.0,7.0
253676,2.0,1.0,1.0,1.0,18.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,4.0,0.0,0.0,1.0,11.0,4.0
253677,0.0,0.0,0.0,1.0,28.0,0.0,0.0,0.0,1.0,1.0,0.0,0.0,1.0,0.0,1.0,0.0,0.0,0.0,2.0,2.0
253678,0.0,1.0,0.0,1.0,23.0,0.0,0.0,0.0,0.0,1.0,1.0,0.0,1.0,0.0,3.0,0.0,0.0,0.0,7.0,1.0


In [333]:
# Rechaecking the dimension
diabetes_prediction_2.shape

(229781, 20)

In [334]:
# Checking the coloumns
diabetes_prediction_2.columns

Index(['Diabetes_012', 'HighBP', 'HighChol', 'CholCheck', 'BMI', 'Smoker',
       'Stroke', 'HeartDiseaseorAttack', 'PhysActivity', 'Fruits', 'Veggies',
       'HvyAlcoholConsump', 'AnyHealthcare', 'NoDocbcCost', 'GenHlth',
       'MentHlth', 'PhysHlth', 'DiffWalk', 'Age', 'Income'],
      dtype='str')

In [335]:
# Checking the number of unique features
diabetes_prediction_2.nunique()

Diabetes_012             3
HighBP                   2
HighChol                 2
CholCheck                2
BMI                     84
Smoker                   2
Stroke                   2
HeartDiseaseorAttack     2
PhysActivity             2
Fruits                   2
Veggies                  2
HvyAlcoholConsump        2
AnyHealthcare            2
NoDocbcCost              2
GenHlth                  5
MentHlth                31
PhysHlth                31
DiffWalk                 2
Age                     13
Income                   8
dtype: int64

In [336]:
# Expression to check categorical features 
features = [col for col in diabetes_prediction_2.columns if diabetes_prediction_2[col].nunique() < 10 and diabetes_prediction_2[col].nunique() > 1]
print(features)

# # Using expression approach to check for categorical features
# features =[col for col in diabetes_dataset_1.columns if diabetes_dataset_1[col].nunique()  < 30 and diabetes_dataset_1[col].nunique() > 1]
# print(features)

['Diabetes_012', 'HighBP', 'HighChol', 'CholCheck', 'Smoker', 'Stroke', 'HeartDiseaseorAttack', 'PhysActivity', 'Fruits', 'Veggies', 'HvyAlcoholConsump', 'AnyHealthcare', 'NoDocbcCost', 'GenHlth', 'DiffWalk', 'Income']


In [337]:
# Loop to check for categorical features
features = []
for col in diabetes_prediction_2.columns.to_list():
    if (diabetes_prediction_2[col].nunique()< 10) and (diabetes_prediction_2[col].nunique()>1):
        features.append(col)
features

['Diabetes_012',
 'HighBP',
 'HighChol',
 'CholCheck',
 'Smoker',
 'Stroke',
 'HeartDiseaseorAttack',
 'PhysActivity',
 'Fruits',
 'Veggies',
 'HvyAlcoholConsump',
 'AnyHealthcare',
 'NoDocbcCost',
 'GenHlth',
 'DiffWalk',
 'Income']

In [338]:
print(diabetes_prediction_2["Diabetes_012"].unique())
print(diabetes_prediction_2["HighBP"].unique())
print(diabetes_prediction_2["HighChol"].unique())
print(diabetes_prediction_2["CholCheck"].unique())
print(diabetes_prediction_2["Smoker"].unique())
print(diabetes_prediction_2["Stroke"].unique())
print(diabetes_prediction_2["HeartDiseaseorAttack"].unique())
print(diabetes_prediction_2["Fruits"].unique())
print(diabetes_prediction_2["Veggies"].unique())
print(diabetes_prediction_2["HvyAlcoholConsump"].unique())
print(diabetes_prediction_2["AnyHealthcare"].unique())
print(diabetes_prediction_2["NoDocbcCost"].unique())
print(diabetes_prediction_2["GenHlth"].unique())
print(diabetes_prediction_2["DiffWalk"].unique())
print(diabetes_prediction_2["Income"].unique())

[0. 2. 1.]
[1. 0.]
[1. 0.]
[1. 0.]
[1. 0.]
[0. 1.]
[0. 1.]
[0. 1.]
[1. 0.]
[0. 1.]
[1. 0.]
[0. 1.]
[5. 3. 2. 4. 1.]
[1. 0.]
[3. 1. 8. 6. 4. 7. 2. 5.]


In [339]:
# Changing the data type of features to category
diabetes_prediction_2[features] = diabetes_prediction_2[features].astype("category")
diabetes_prediction_2.info()

<class 'pandas.DataFrame'>
Index: 229781 entries, 0 to 253679
Data columns (total 20 columns):
 #   Column                Non-Null Count   Dtype   
---  ------                --------------   -----   
 0   Diabetes_012          229781 non-null  category
 1   HighBP                229781 non-null  category
 2   HighChol              229781 non-null  category
 3   CholCheck             229781 non-null  category
 4   BMI                   229781 non-null  float64 
 5   Smoker                229781 non-null  category
 6   Stroke                229781 non-null  category
 7   HeartDiseaseorAttack  229781 non-null  category
 8   PhysActivity          229781 non-null  category
 9   Fruits                229781 non-null  category
 10  Veggies               229781 non-null  category
 11  HvyAlcoholConsump     229781 non-null  category
 12  AnyHealthcare         229781 non-null  category
 13  NoDocbcCost           229781 non-null  category
 14  GenHlth               229781 non-null  category
 15 

The dataset originally contained 253,680 rows and 22 columns. A new copy of the dataset was created, after removing 23,899 duplicate rows, the new dataset was reduced to 229,781 rows. Two columns (Education and Sex) were dropped, leaving 20 columns. No missing values were found. The dataset is entirely numerical, originally stored as float64. Sixteen columns were converted to category type.

In [340]:
# importing the diabetes dataset csv 
diabetes_dataset = pd.read_csv("diabetes_dataset.csv")
diabetes_dataset

,age,gender,ethnicity,education_level,income_level,employment_status,smoking_status,alcohol_consumption_per_week,physical_activity_minutes_per_week,diet_score,...,hdl_cholesterol,ldl_cholesterol,triglycerides,glucose_fasting,glucose_postprandial,insulin_level,hba1c,diabetes_risk_score,diabetes_stage,diagnosed_diabetes
0,58,Male,Asian,Highschool,Lower-Middle,Employed,Never,0,215,5.7,...,41,160,145,136,236,6.36,8.18,29.6,Type 2,1
1,48,Female,White,Highschool,Middle,Employed,Former,1,143,6.7,...,55,50,30,93,150,2.00,5.63,23.0,No Diabetes,0
2,60,Male,Hispanic,Highschool,Middle,Unemployed,Never,1,57,6.4,...,66,99,36,118,195,5.07,7.51,44.7,Type 2,1
3,74,Female,Black,Highschool,Low,Retired,Never,0,49,3.4,...,50,79,140,139,253,5.28,9.03,38.2,Type 2,1
4,46,Male,White,Graduate,Middle,Retired,Never,1,109,7.2,...,52,125,160,137,184,12.74,7.20,23.5,Type 2,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
99995,46,Male,Other,Graduate,Upper-Middle,Unemployed,Former,1,136,8.3,...,45,150,116,113,109,14.58,5.55,26.0,Pre-Diabetes,0
99996,41,Female,White,Graduate,Middle,Employed,Never,3,76,8.8,...,55,123,146,96,146,9.02,5.97,24.4,Pre-Diabetes,0
99997,57,Female,Black,No formal,Upper-Middle,Employed,Former,4,121,9.9,...,50,111,184,93,132,2.57,5.21,27.6,No Diabetes,0
99998,47,Female,Black,Highschool,Lower-Middle,Retired,Never,3,52,5.9,...,68,91,116,106,117,9.81,5.53,26.4,Pre-Diabetes,0


In [341]:
# Getting the first 5 rows
diabetes_dataset.head(5)

,age,gender,ethnicity,education_level,income_level,employment_status,smoking_status,alcohol_consumption_per_week,physical_activity_minutes_per_week,diet_score,...,hdl_cholesterol,ldl_cholesterol,triglycerides,glucose_fasting,glucose_postprandial,insulin_level,hba1c,diabetes_risk_score,diabetes_stage,diagnosed_diabetes
0,58,Male,Asian,Highschool,Lower-Middle,Employed,Never,0,215,5.7,...,41,160,145,136,236,6.36,8.18,29.6,Type 2,1
1,48,Female,White,Highschool,Middle,Employed,Former,1,143,6.7,...,55,50,30,93,150,2.00,5.63,23.0,No Diabetes,0
2,60,Male,Hispanic,Highschool,Middle,Unemployed,Never,1,57,6.4,...,66,99,36,118,195,5.07,7.51,44.7,Type 2,1
3,74,Female,Black,Highschool,Low,Retired,Never,0,49,3.4,...,50,79,140,139,253,5.28,9.03,38.2,Type 2,1
4,46,Male,White,Graduate,Middle,Retired,Never,1,109,7.2,...,52,125,160,137,184,12.74,7.20,23.5,Type 2,1


In [342]:
# Checking the dataset information
diabetes_dataset.info()

<class 'pandas.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 31 columns):
 #   Column                              Non-Null Count   Dtype  
---  ------                              --------------   -----  
 0   age                                 100000 non-null  int64  
 1   gender                              100000 non-null  str    
 2   ethnicity                           100000 non-null  str    
 3   education_level                     100000 non-null  str    
 4   income_level                        100000 non-null  str    
 5   employment_status                   100000 non-null  str    
 6   smoking_status                      100000 non-null  str    
 7   alcohol_consumption_per_week        100000 non-null  int64  
 8   physical_activity_minutes_per_week  100000 non-null  int64  
 9   diet_score                          100000 non-null  float64
 10  sleep_hours_per_day                 100000 non-null  float64
 11  screen_time_hours_per_day           10

In [343]:
# Checking the dimensions of the dataset
diabetes_dataset.shape

(100000, 31)

In [344]:
# Checking for null values
diabetes_dataset.isnull().sum()

age                                   0
gender                                0
ethnicity                             0
education_level                       0
income_level                          0
employment_status                     0
smoking_status                        0
alcohol_consumption_per_week          0
physical_activity_minutes_per_week    0
diet_score                            0
sleep_hours_per_day                   0
screen_time_hours_per_day             0
family_history_diabetes               0
hypertension_history                  0
cardiovascular_history                0
bmi                                   0
waist_to_hip_ratio                    0
systolic_bp                           0
diastolic_bp                          0
heart_rate                            0
cholesterol_total                     0
hdl_cholesterol                       0
ldl_cholesterol                       0
triglycerides                         0
glucose_fasting                       0


In [345]:
# Checking for duplicates
print(diabetes_dataset.duplicated().sum())

0


In [346]:
# Copying the original dataset
diabetes_dataset_1 = diabetes_dataset.copy()
diabetes_dataset_1

,age,gender,ethnicity,education_level,income_level,employment_status,smoking_status,alcohol_consumption_per_week,physical_activity_minutes_per_week,diet_score,...,hdl_cholesterol,ldl_cholesterol,triglycerides,glucose_fasting,glucose_postprandial,insulin_level,hba1c,diabetes_risk_score,diabetes_stage,diagnosed_diabetes
0,58,Male,Asian,Highschool,Lower-Middle,Employed,Never,0,215,5.7,...,41,160,145,136,236,6.36,8.18,29.6,Type 2,1
1,48,Female,White,Highschool,Middle,Employed,Former,1,143,6.7,...,55,50,30,93,150,2.00,5.63,23.0,No Diabetes,0
2,60,Male,Hispanic,Highschool,Middle,Unemployed,Never,1,57,6.4,...,66,99,36,118,195,5.07,7.51,44.7,Type 2,1
3,74,Female,Black,Highschool,Low,Retired,Never,0,49,3.4,...,50,79,140,139,253,5.28,9.03,38.2,Type 2,1
4,46,Male,White,Graduate,Middle,Retired,Never,1,109,7.2,...,52,125,160,137,184,12.74,7.20,23.5,Type 2,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
99995,46,Male,Other,Graduate,Upper-Middle,Unemployed,Former,1,136,8.3,...,45,150,116,113,109,14.58,5.55,26.0,Pre-Diabetes,0
99996,41,Female,White,Graduate,Middle,Employed,Never,3,76,8.8,...,55,123,146,96,146,9.02,5.97,24.4,Pre-Diabetes,0
99997,57,Female,Black,No formal,Upper-Middle,Employed,Former,4,121,9.9,...,50,111,184,93,132,2.57,5.21,27.6,No Diabetes,0
99998,47,Female,Black,Highschool,Lower-Middle,Retired,Never,3,52,5.9,...,68,91,116,106,117,9.81,5.53,26.4,Pre-Diabetes,0


In [347]:
# Dropping columns
diabetes_dataset_1.drop(columns = ["gender", "education_level", "employment_status"], inplace=True)
diabetes_dataset_1

,age,ethnicity,income_level,smoking_status,alcohol_consumption_per_week,physical_activity_minutes_per_week,diet_score,sleep_hours_per_day,screen_time_hours_per_day,family_history_diabetes,...,hdl_cholesterol,ldl_cholesterol,triglycerides,glucose_fasting,glucose_postprandial,insulin_level,hba1c,diabetes_risk_score,diabetes_stage,diagnosed_diabetes
0,58,Asian,Lower-Middle,Never,0,215,5.7,7.9,7.9,0,...,41,160,145,136,236,6.36,8.18,29.6,Type 2,1
1,48,White,Middle,Former,1,143,6.7,6.5,8.7,0,...,55,50,30,93,150,2.00,5.63,23.0,No Diabetes,0
2,60,Hispanic,Middle,Never,1,57,6.4,10.0,8.1,1,...,66,99,36,118,195,5.07,7.51,44.7,Type 2,1
3,74,Black,Low,Never,0,49,3.4,6.6,5.2,0,...,50,79,140,139,253,5.28,9.03,38.2,Type 2,1
4,46,White,Middle,Never,1,109,7.2,7.4,5.0,0,...,52,125,160,137,184,12.74,7.20,23.5,Type 2,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
99995,46,Other,Upper-Middle,Former,1,136,8.3,6.4,4.5,0,...,45,150,116,113,109,14.58,5.55,26.0,Pre-Diabetes,0
99996,41,White,Middle,Never,3,76,8.8,6.8,4.8,0,...,55,123,146,96,146,9.02,5.97,24.4,Pre-Diabetes,0
99997,57,Black,Upper-Middle,Former,4,121,9.9,5.0,6.1,0,...,50,111,184,93,132,2.57,5.21,27.6,No Diabetes,0
99998,47,Black,Lower-Middle,Never,3,52,5.9,6.7,0.5,0,...,68,91,116,106,117,9.81,5.53,26.4,Pre-Diabetes,0


In [348]:
# Rechecking dimension
diabetes_dataset_1.shape

(100000, 28)

In [349]:
# Getting the first 5 rows
diabetes_dataset_1.head(5)

,age,ethnicity,income_level,smoking_status,alcohol_consumption_per_week,physical_activity_minutes_per_week,diet_score,sleep_hours_per_day,screen_time_hours_per_day,family_history_diabetes,...,hdl_cholesterol,ldl_cholesterol,triglycerides,glucose_fasting,glucose_postprandial,insulin_level,hba1c,diabetes_risk_score,diabetes_stage,diagnosed_diabetes
0,58,Asian,Lower-Middle,Never,0,215,5.7,7.9,7.9,0,...,41,160,145,136,236,6.36,8.18,29.6,Type 2,1
1,48,White,Middle,Former,1,143,6.7,6.5,8.7,0,...,55,50,30,93,150,2.00,5.63,23.0,No Diabetes,0
2,60,Hispanic,Middle,Never,1,57,6.4,10.0,8.1,1,...,66,99,36,118,195,5.07,7.51,44.7,Type 2,1
3,74,Black,Low,Never,0,49,3.4,6.6,5.2,0,...,50,79,140,139,253,5.28,9.03,38.2,Type 2,1
4,46,White,Middle,Never,1,109,7.2,7.4,5.0,0,...,52,125,160,137,184,12.74,7.20,23.5,Type 2,1


In [350]:
# Rechecking dimension
diabetes_dataset_1.info()

<class 'pandas.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 28 columns):
 #   Column                              Non-Null Count   Dtype  
---  ------                              --------------   -----  
 0   age                                 100000 non-null  int64  
 1   ethnicity                           100000 non-null  str    
 2   income_level                        100000 non-null  str    
 3   smoking_status                      100000 non-null  str    
 4   alcohol_consumption_per_week        100000 non-null  int64  
 5   physical_activity_minutes_per_week  100000 non-null  int64  
 6   diet_score                          100000 non-null  float64
 7   sleep_hours_per_day                 100000 non-null  float64
 8   screen_time_hours_per_day           100000 non-null  float64
 9   family_history_diabetes             100000 non-null  int64  
 10  hypertension_history                100000 non-null  int64  
 11  cardiovascular_history              10

In [351]:
# Checking for duplicate values
print(diabetes_dataset_1.columns.to_list())

['age', 'ethnicity', 'income_level', 'smoking_status', 'alcohol_consumption_per_week', 'physical_activity_minutes_per_week', 'diet_score', 'sleep_hours_per_day', 'screen_time_hours_per_day', 'family_history_diabetes', 'hypertension_history', 'cardiovascular_history', 'bmi', 'waist_to_hip_ratio', 'systolic_bp', 'diastolic_bp', 'heart_rate', 'cholesterol_total', 'hdl_cholesterol', 'ldl_cholesterol', 'triglycerides', 'glucose_fasting', 'glucose_postprandial', 'insulin_level', 'hba1c', 'diabetes_risk_score', 'diabetes_stage', 'diagnosed_diabetes']


In [352]:
# Getting the names of colomns in diabetes dataset data
print(diabetes_dataset_1.columns)

Index(['age', 'ethnicity', 'income_level', 'smoking_status',
       'alcohol_consumption_per_week', 'physical_activity_minutes_per_week',
       'diet_score', 'sleep_hours_per_day', 'screen_time_hours_per_day',
       'family_history_diabetes', 'hypertension_history',
       'cardiovascular_history', 'bmi', 'waist_to_hip_ratio', 'systolic_bp',
       'diastolic_bp', 'heart_rate', 'cholesterol_total', 'hdl_cholesterol',
       'ldl_cholesterol', 'triglycerides', 'glucose_fasting',
       'glucose_postprandial', 'insulin_level', 'hba1c', 'diabetes_risk_score',
       'diabetes_stage', 'diagnosed_diabetes'],
      dtype='str')


In [353]:
# Checking the number of unique features
diabetes_dataset_1.nunique()

age                                     73
ethnicity                                5
income_level                             5
smoking_status                           3
alcohol_consumption_per_week            11
physical_activity_minutes_per_week     620
diet_score                             101
sleep_hours_per_day                     71
screen_time_hours_per_day              156
family_history_diabetes                  2
hypertension_history                     2
cardiovascular_history                   2
bmi                                    240
waist_to_hip_ratio                      40
systolic_bp                             86
diastolic_bp                            60
heart_rate                              64
cholesterol_total                      210
hdl_cholesterol                         79
ldl_cholesterol                        190
triglycerides                          262
glucose_fasting                        109
glucose_postprandial                   210
insulin_lev

In [354]:
# Using expression approach to check for categorical features
features =[col for col in diabetes_dataset_1.columns if diabetes_dataset_1[col].nunique()  < 30 and diabetes_dataset_1[col].nunique() > 1]
print(features)

['ethnicity', 'income_level', 'smoking_status', 'alcohol_consumption_per_week', 'family_history_diabetes', 'hypertension_history', 'cardiovascular_history', 'diabetes_stage', 'diagnosed_diabetes']


In [355]:
# loop approach for checking categorical features
features = []
for col in diabetes_dataset_1.columns.to_list():
    if (diabetes_dataset_1[col].nunique()< 30) and (diabetes_dataset_1[col].nunique()>1):
        features.append(col)
features

['ethnicity',
 'income_level',
 'smoking_status',
 'alcohol_consumption_per_week',
 'family_history_diabetes',
 'hypertension_history',
 'cardiovascular_history',
 'diabetes_stage',
 'diagnosed_diabetes']

In [356]:
# Checking the dataset information
diabetes_dataset_1.info()

<class 'pandas.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 28 columns):
 #   Column                              Non-Null Count   Dtype  
---  ------                              --------------   -----  
 0   age                                 100000 non-null  int64  
 1   ethnicity                           100000 non-null  str    
 2   income_level                        100000 non-null  str    
 3   smoking_status                      100000 non-null  str    
 4   alcohol_consumption_per_week        100000 non-null  int64  
 5   physical_activity_minutes_per_week  100000 non-null  int64  
 6   diet_score                          100000 non-null  float64
 7   sleep_hours_per_day                 100000 non-null  float64
 8   screen_time_hours_per_day           100000 non-null  float64
 9   family_history_diabetes             100000 non-null  int64  
 10  hypertension_history                100000 non-null  int64  
 11  cardiovascular_history              10

In [357]:
print(diabetes_dataset_1["diagnosed_diabetes"].unique())
print(diabetes_dataset_1["diabetes_stage"].unique())
print(diabetes_dataset_1["cardiovascular_history"].unique())
print(diabetes_dataset_1["hypertension_history"].unique())
print(diabetes_dataset_1["family_history_diabetes"].unique())
print(diabetes_dataset_1["alcohol_consumption_per_week"].unique())
print(diabetes_dataset_1["smoking_status"].unique())
print(diabetes_dataset_1["income_level"].unique())
print(diabetes_dataset_1["ethnicity"].unique())

[1 0]
<StringArray>
['Type 2', 'No Diabetes', 'Pre-Diabetes', 'Gestational', 'Type 1']
Length: 5, dtype: str
[0 1]
[0 1]
[0 1]
[ 0  1  2  3  6  5  4  9  8  7 10]
<StringArray>
['Never', 'Former', 'Current']
Length: 3, dtype: str
<StringArray>
['Lower-Middle', 'Middle', 'Low', 'Upper-Middle', 'High']
Length: 5, dtype: str
<StringArray>
['Asian', 'White', 'Hispanic', 'Black', 'Other']
Length: 5, dtype: str


In [358]:
# Changing the data type of features to category
diabetes_dataset_1[features] = diabetes_dataset_1[features].astype("category")
diabetes_dataset_1.info()

<class 'pandas.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 28 columns):
 #   Column                              Non-Null Count   Dtype   
---  ------                              --------------   -----   
 0   age                                 100000 non-null  int64   
 1   ethnicity                           100000 non-null  category
 2   income_level                        100000 non-null  category
 3   smoking_status                      100000 non-null  category
 4   alcohol_consumption_per_week        100000 non-null  category
 5   physical_activity_minutes_per_week  100000 non-null  int64   
 6   diet_score                          100000 non-null  float64 
 7   sleep_hours_per_day                 100000 non-null  float64 
 8   screen_time_hours_per_day           100000 non-null  float64 
 9   family_history_diabetes             100000 non-null  category
 10  hypertension_history                100000 non-null  category
 11  cardiovascular_history   

In [359]:
# Making another copy of the dataset
diabetes_confirmation = diabetes_dataset_1.copy()
diabetes_confirmation

,age,ethnicity,income_level,smoking_status,alcohol_consumption_per_week,physical_activity_minutes_per_week,diet_score,sleep_hours_per_day,screen_time_hours_per_day,family_history_diabetes,...,hdl_cholesterol,ldl_cholesterol,triglycerides,glucose_fasting,glucose_postprandial,insulin_level,hba1c,diabetes_risk_score,diabetes_stage,diagnosed_diabetes
0,58,Asian,Lower-Middle,Never,0,215,5.7,7.9,7.9,0,...,41,160,145,136,236,6.36,8.18,29.6,Type 2,1
1,48,White,Middle,Former,1,143,6.7,6.5,8.7,0,...,55,50,30,93,150,2.00,5.63,23.0,No Diabetes,0
2,60,Hispanic,Middle,Never,1,57,6.4,10.0,8.1,1,...,66,99,36,118,195,5.07,7.51,44.7,Type 2,1
3,74,Black,Low,Never,0,49,3.4,6.6,5.2,0,...,50,79,140,139,253,5.28,9.03,38.2,Type 2,1
4,46,White,Middle,Never,1,109,7.2,7.4,5.0,0,...,52,125,160,137,184,12.74,7.20,23.5,Type 2,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
99995,46,Other,Upper-Middle,Former,1,136,8.3,6.4,4.5,0,...,45,150,116,113,109,14.58,5.55,26.0,Pre-Diabetes,0
99996,41,White,Middle,Never,3,76,8.8,6.8,4.8,0,...,55,123,146,96,146,9.02,5.97,24.4,Pre-Diabetes,0
99997,57,Black,Upper-Middle,Former,4,121,9.9,5.0,6.1,0,...,50,111,184,93,132,2.57,5.21,27.6,No Diabetes,0
99998,47,Black,Lower-Middle,Never,3,52,5.9,6.7,0.5,0,...,68,91,116,106,117,9.81,5.53,26.4,Pre-Diabetes,0


The dataset contains 100,000 rows and 28 columns after removing 3 columns (gender, education_level, employment_status). All 28 columns are non-null with no missing values found across the entire dataset. No duplicate rows were detected. Six binary columns were converted to category type. 

In [360]:
# dropping confimation status colomns
diabetes_confirmation.drop(columns = ["glucose_fasting", "glucose_postprandial", "insulin_level", "hba1c"], inplace = True)
diabetes_confirmation

,age,ethnicity,income_level,smoking_status,alcohol_consumption_per_week,physical_activity_minutes_per_week,diet_score,sleep_hours_per_day,screen_time_hours_per_day,family_history_diabetes,...,systolic_bp,diastolic_bp,heart_rate,cholesterol_total,hdl_cholesterol,ldl_cholesterol,triglycerides,diabetes_risk_score,diabetes_stage,diagnosed_diabetes
0,58,Asian,Lower-Middle,Never,0,215,5.7,7.9,7.9,0,...,134,78,68,239,41,160,145,29.6,Type 2,1
1,48,White,Middle,Former,1,143,6.7,6.5,8.7,0,...,129,76,67,116,55,50,30,23.0,No Diabetes,0
2,60,Hispanic,Middle,Never,1,57,6.4,10.0,8.1,1,...,115,73,74,213,66,99,36,44.7,Type 2,1
3,74,Black,Low,Never,0,49,3.4,6.6,5.2,0,...,120,93,68,171,50,79,140,38.2,Type 2,1
4,46,White,Middle,Never,1,109,7.2,7.4,5.0,0,...,92,67,67,210,52,125,160,23.5,Type 2,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
99995,46,Other,Upper-Middle,Former,1,136,8.3,6.4,4.5,0,...,126,74,63,227,45,150,116,26.0,Pre-Diabetes,0
99996,41,White,Middle,Never,3,76,8.8,6.8,4.8,0,...,130,76,57,207,55,123,146,24.4,Pre-Diabetes,0
99997,57,Black,Upper-Middle,Former,4,121,9.9,5.0,6.1,0,...,107,87,69,189,50,111,184,27.6,No Diabetes,0
99998,47,Black,Lower-Middle,Never,3,52,5.9,6.7,0.5,0,...,115,77,58,173,68,91,116,26.4,Pre-Diabetes,0


In [361]:
diabetes_confirmation.info()

<class 'pandas.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 24 columns):
 #   Column                              Non-Null Count   Dtype   
---  ------                              --------------   -----   
 0   age                                 100000 non-null  int64   
 1   ethnicity                           100000 non-null  category
 2   income_level                        100000 non-null  category
 3   smoking_status                      100000 non-null  category
 4   alcohol_consumption_per_week        100000 non-null  category
 5   physical_activity_minutes_per_week  100000 non-null  int64   
 6   diet_score                          100000 non-null  float64 
 7   sleep_hours_per_day                 100000 non-null  float64 
 8   screen_time_hours_per_day           100000 non-null  float64 
 9   family_history_diabetes             100000 non-null  category
 10  hypertension_history                100000 non-null  category
 11  cardiovascular_history   

In [362]:
diabetes_confirmation.shape

(100000, 24)

A new copy of the dataset was created to remove features that are strictly for diabetes confirmation and not prediction.